File: backend/src/main/java/com/petwellness/model/Appointment.java
package com.petwellness.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

/**
 * Model representing a booked appointment by a user.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "appointments")
public class Appointment {

    @Id
    private String id;

    @Indexed
    private String userId;

    @Indexed
    private String petId;

    @Indexed
    private String slotId;

    private LocalDate appointmentDate;

    private LocalTime appointmentTime;

    private ConsultationType consultationType;

    private String veterinarianName;

    private AppointmentStatus status;

    private String notes;

    @CreatedDate
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}

File: frontend/src/types/appointment.ts
export const ConsultationType = {
    ONLINE: 'ONLINE',
    IN_CLINIC: 'IN_CLINIC',
} as const;

export type ConsultationType = (typeof ConsultationType)[keyof typeof ConsultationType];

export const SlotStatus = {
    AVAILABLE: 'AVAILABLE',
    BOOKED: 'BOOKED',
    CANCELLED: 'CANCELLED',
} as const;

export type SlotStatus = (typeof SlotStatus)[keyof typeof SlotStatus];

export const AppointmentStatus = {
    SCHEDULED: 'SCHEDULED',
    COMPLETED: 'COMPLETED',
    CANCELLED: 'CANCELLED',
} as const;

export type AppointmentStatus = (typeof AppointmentStatus)[keyof typeof AppointmentStatus];

export interface AppointmentSlotResponse {
    id: string;
    date: string;
    time: string;
    consultationType: ConsultationType;
    veterinarianName: string;
    status: SlotStatus;
    duration: number;
    consultationFee?: number;
    createdAt: string;
    updatedAt: string;
}

export interface BookAppointmentRequest {
    petId: string;
    slotId: string;
    notes?: string;
}

export interface AppointmentResponse {
    id: string;
    userId: string;
    petId: string;
    slotId: string;
    appointmentDate: string;
    appointmentTime: string;
    consultationType: ConsultationType;
    veterinarianName: string;
    status: AppointmentStatus;
    notes: string;
    createdAt: string;
    updatedAt: string;
}
